tests/tutor/test_tutor_teacher.py

```diff
@@ -0,0 +1,1330 @@
+    # GIVEN: Logged into Tutor as a teacher
+    # AND: Has an existing course with an external assignment
+
+    # WHEN: Click on an external assignment.
+    # AND: Change all the required fields
+    # AND: Click “cancel""
+
+    # THEN: User is taken back to the Calender
+    # AND: No changes should be made on the assignment
+
+
+@test_case('')
+@tutor
+def test_cancel_editing_an_external_draft(tutor_base_url, selenium, teacher):
+    """Cancel editing an external assignment draft as teacher."""
+    # GIVEN: Logged into Tutor as a teacher
+    # AND: Has an existing course with an external assignment draft
+
+    # WHEN: Click on an external assignment draft
+    # AND: Change all the required fields
+    # AND: Click “cancel"
+
+    # THEN: User is taken back to the Calender
+    # AND: No changes should be made on the draft
+
+
+@test_case('')
+@tutor
+def test_adding_external_assignment_with_empty_fields(tutor_base_url, selenium,
+                                                      teacher):
+    """Cancel editing an external assignment as teacher."""
+    # GIVEN: Logged into Tutor as a teacher
+    # AND: Has an existing course
+
+    # WHEN: Click on an external assignment
+    # AND: Change some of the required fields and leave some blank
+    # AND: Click "Publish"
+
+    # THEN: Red outlines should popup for required fields
+
+
+@test_case('')
+@tutor
+def test_add_a_new_event(tutor_base_url, selenium, teacher):
+    """Adding a new event as teacher."""
+    # GIVEN: Logged into Tutor as a teacher
+    # AND: Has an existing course
+
+    # WHEN: Go to the course page
+    # AND: Click on a date on the dashboard
+    # AND: Click ""add event""
+    # AND: Fill in all the required fields
+    # AND: click "Publish"
+
+    # THEN: User is taken back to the Calender
+    # AND: A new event should be visible on the calendar
+
+
+@test_case('')
+@tutor
+def test_save_an_event_draft(tutor_base_url, selenium, teacher):
+    """Saving a draft event as teacher."""
+    # GIVEN: Logged into Tutor as a teacher
+    # AND: Has an existing course
+
+    # WHEN: click on a date on the dashboard, then click ""add event""
+    # AND: fill in all the required fields
+    # AND: click "Save as Draft"
+
+    # THEN: User is taken back to the Calender
+    # AND: A new event draft should be visible on the calendar
+
+
+@test_case('')
+@tutor
+def test_edit_a_published_event(tutor_base_url, selenium, teacher):
+    """Editing a published event as teacher."""
+    # GIVEN: Logged into Tutor as a teacher
+    # AND: Has an existing course
+    # AND: Has an published event
+
+    # WHEN: Click on a published event
+    # AND: Change all the required fields
+    # AND: Click "Publish"
+
+    # THEN: User is taken back to the Calender
+    # AND: The event should be visible on the calendar with its info updated"
+
+
+@test_case('')
+@tutor
+def test_edit_an_event_draft(tutor_base_url, selenium, teacher):
+    """editing an draft event as teacher."""
```

Comment:
Watch out for the imperative voice and capitalization...
'editing an draft event as teacher.' should be 'Edit a draft event.' That way it is a directed action; the user type is implied by both the `teacher` argument as well as the action (can only be performed by an instructor).